presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -687,6 +707,53 @@ private static URI findCancelableLeafStage(StageInfo stage)
         return stage.getSelf();
     }
 
+    private boolean planHashesEqual(Optional<CanonicalPlanWithInfo> currentTopLevelPlanHash)
```

Comment:
This is a relatively simple function and only called once, may not need to be a separate function